src/node/rpc/jwt_management.h

```diff
@@ -113,6 +173,16 @@ namespace ccf
         return false;
       }
 
+      if (keys->has(kid) && der != keys->get(kid))
```

Comment:
Agree with log, but not sure about early false here. If this happens, I think we want it flagged, but we don't necessarily want to stop updating keys.

Reply:
That's what we did for other if-blocks around. E.g.:

```
      if (keys->has(kid) && key_issuer->get(kid).value() != issuer)
      {
        LOG_FAIL_FMT(
          "{}: key id {} already added for different issuer", log_prefix, kid);
        return false;
      }

```

I'm inclined to preserve that because the call is "set_keys", so setting them partially and returning `true` doesn't seem right.

Let me know your thoughts if you insist we should consider changing it.

Reply:
Agree we should not do a partial set in any circumstances. But I think we could replace the mapping altogether rather than back off.

Reply:
By avoiding partial set for the current issuer being set, we do so for the previous one which has a conflicting keys, don't we?

In this particular case we do overwrite a key for some issuer, which may be only one of the keys that issuer has set.

Moreover, we now introduce new inconsistency, because if we do decide to simply overwrite - the issuer for previous kid is kept in issuers map.

Reply:
I meant the mapping for the issuer. There are two possible collisions here:

1. A kid collision for a single issuer. This amounts to an inconsistent update set from that issuer, and a violation of the OpenID spec. Rejecting the update wholesale is clearly the only coherent behaviour.
2. A kid collision between two issuer. Two coherent behaviours are possible: reject the update, or accept it entirely and remove the keys for the issuer that had colliding keys (i.e. latest wins, as opposed to first wins).

I am not sure that we need to make this trade-off though, because it seems to me that we do need to have a unique `kid -> pem`, and we could instead have a `kid -> [pem, iss]`, which we try to validate one by one until we get a match. In fact, I don't understand how we really solve #5177 with this check.

Reply:
> In fact, I don't understand how we really solve https://github.com/microsoft/CCF/issues/5177 with this check.

1. Scenarios like this are not permitted anymore
```
- Iss A set [(kid_1, pem_1), (kid_2, pem_2)]
- Iss B set [(kid_1, pem_0), (kid_3, pem_3)]
- Remove Iss B
```
Resulting kids -> pem map
kid_1 -> pem_0
kid_2 -> pem_2

2 (my opinion): Explicit > implicit, so a silent key overwriting has been replaced by conflicting key rejection

Reply:
>  we could instead have a kid -> [pem, iss], which we try to validate one by one until we get a match
I agree, and I'd rather implement this instead.

The reason we decided not to do it after discussing within a team was twofold
- key validation is expensive, doing several might affect the performance
- overcomplicates the code, with a rather tine chance to happen in practice

So the decision was to leave it as is and if implement only if someone really needs it.

Options we have
- implement it here
- create a ticket and implement on demand later on

I prefer the latter, but I'll take the one you suggest, because I have little experience of practical application of CCF.